src/Compilers/CSharp/Portable/Binder/Binder_AnonymousTypes.cs

```diff
@@ -192,19 +192,7 @@ private TypeSymbol GetAnonymousTypeFieldType(BoundExpression expression, CSharpS
                 }
                 else
                 {
-                    if (expression.Kind == BoundKind.UnboundLambda)
-                    {
-                        errorArg = ((UnboundLambda)expression).MessageID.Localize();
-                    }
-                    else if (expression.Kind == BoundKind.MethodGroup)
-                    {
-                        errorArg = MessageID.IDS_MethodGroup.Localize();
-                    }
-                    else
-                    {
-                        Debug.Assert(expression.IsLiteralNull(), "How did we successfully bind an expression without a type?");
-                        errorArg = MessageID.IDS_NULL.Localize();
-                    }
+                    errorArg = expression.Display;
```

Comment:
It is related to the feature work, as I don't want to have to add a case for `BoundKind.ThrowExpression` everywhere this code appears. `expression.Display` was previously added for just this purpose, it just wasn't rolled out to all the sites where it should have been.